packages/api-server/src/requestHandlers/awsLambdaFastify.ts

```diff
@@ -32,14 +33,34 @@ const fastifyResponseForLambdaResult = (
 ) => {
   const {
     statusCode = 200,
-    headers,
+    headers = {},
     body = '',
-    multiValueHeaders,
+    multiValueHeaders = {},
   } = lambdaResult
   const h = mergeMultiValueHeaders(headers, multiValueHeaders)
-  reply.headers(h)
+  const headersWithoutCookie = omit(h, 'set-cookie')
+  reply.headers(headersWithoutCookie)
   reply.status(statusCode)
 
+  console.log(mergeMultiValueHeaders, headers)
+
+  Object.keys(multiValueHeaders).forEach((key) => {
+    const isSetCookie = key.toLowerCase() === 'set-cookie'
```

Comment:
If you took this part of the code Line 48 to 54, and put it in the mergeMultiValueHeaders function? 

This function could potentially return an object `{mergedHeaders, cookieHeaders}` which you can then do something like:

```
reply.headers(mergedHeaders)

if (cookieHeaders) {
  cookieHeaders.forEach(cookieHeader => {
		reply.header('set-cookie', cookieHeader)
  })
}
```

Reply:
This way you will also be able to add a test to packages/api-server/src/__tests__/requestHandlers/utils.test.ts with the cookie usecases